### PR TITLE
Add ``gpu-examples`` image

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR adds a new `coiled/gpu-examples` image, which is the existing `coiled/gpu` image + the examples in this repo. This will be convenient for getting started with GPU examples

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -17,4 +17,8 @@ RUN mamba env update -n base --file pytorch.yml \
         torchvision \
         "pytorch-cuda=${CUDA_VER%.*}.*" \
     && conda clean -afy \
-    && rm pytorch.yml
\ No newline at end of file
+    && rm pytorch.yml
+
+
+FROM base as examples
+COPY . .
```
Comment (jrbourbeau) [inline Dockerfile:24]: IIUC the downside of this `COPY` is that we would want to rebuild this image as notebooks change (which is more than we currently do). I'm personally fine with this. 

An alternative, again IIUC, would be to make this a `git clone ...` where we pull whatever the current status of the repo is. @ntabris maybe you have thoughts?

Comment (ntabris) [reply Dockerfile:24]: I was thinking you could just have something like

```
FROM coiled/gpu:latest as base
COPY <stuff you want to copy>
```

Comment (jrbourbeau) [reply Dockerfile:24]: Ah, so a separate `Dockerfile`?

Comment (jrbourbeau) [reply Dockerfile:24]: I'm curious if there's a benefit to having multiple dockerfiles. Just separation of concerns, or maybe something more than that?

Comment (ntabris) [reply Dockerfile:24]: I don't think it matters much, that's just what I'd do. Maybe I'm not following what you were asking originally.

Comment (ntabris) [reply Dockerfile:24]: fwiw, rapids does this: https://github.com/rapidsai/docker/blob/branch-23.10/Dockerfile

**Comment by mrocklin** [comment]
When this is in we'll want to update the Getting Started UX in the app to point to `coiled/gpu-examples`.  cc @shughes-uk

**Comment by jrbourbeau** [comment]
Alright, I'm happy with the current state of this. @ntabris do you have time for a quick review?

**Pull request merged**